angular-app/src/app/create-project-dialog/create-project-dialog.component.spec.ts

```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { CreateProjectDialogComponent } from './create-project-dialog.component';
+
+describe('CreateProjectDialogComponent', () => {
+  let component: CreateProjectDialogComponent;
+  let fixture: ComponentFixture<CreateProjectDialogComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ CreateProjectDialogComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateProjectDialogComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
```

Comment:
Suggestion for future commits: Think about how you can add better tests in future Pull requests. Testing items like validation that your component doesn't try to create a new project in your database unless it has a valid name, would be a good place to start.

Reply:
Yes, testing is a good thing that should be done --- everything currently in the code has been autogenerated. As mentioned before, I looked into it last week and had to do a lot of work just to get it to pass these basic tests because it involves mocking all of the external services (particularly Firestore and FireAuth). I still need to update that branch to reflect new changes and clean it up so that mocks are shared and not copy-pasted into each .spec file; I promise that will be in by the start of next week at the latest so that we can do more comprehensive testing on everything.

Reply:
Sounds good. Setup of the mocks, is the largest work. Once you do that once, subsequent tests should be quick to implement. As a suggestion, given many have probably made mocks for firebase look at outside libraries to see if they can alleviate the upfront setup cost. https://www.npmjs.com/package/firestore-jest-mock. You do not have to use the libraries, but they may be helpful.

Reply:
Thanks! I had been looking up things that were closer to example stubs than mocks, so this seems useful.